### ISSUE TITLE
Response bodies from failed requests should be logged

### ISSUE THREAD

**Issue by darrylring** [elastic/elasticsearch-py#346]
`log_request_success` current logs both the request and response bodies, while `log_request_fail` only logs the request body. An exception may be thrown, but there's nothing in the log beyond the HTTP status code.

This:

```
WARNING:elasticsearch:GET /blah/_search [status:404 request:0.012s]
DEBUG:elasticsearch:> {}
```

Should be:

```
WARNING:elasticsearch:GET /blah/_search [status:404 request:0.012s]
DEBUG:elasticsearch:> {}
DEBUG:elasticsearch:< {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"blah","index":"blah"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"blah","index":"blah"},"status":404}
```

I found that while using date math in index names, I couldn't actually see what the calculated index name was when it couldn't be found.

**Comment by HonzaKral** [comment]
Fixed in #347 
Thanks for the report and the PR!

**Comment by darrylring** [comment]
My pleasure. As you can see, I didn't fail at this commit a bunch of times or anything...

Closing.
